servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/IterableMergeCompletable.java

```diff
@@ -37,49 +35,37 @@
 
     @Override
     public MergeSubscriber apply(final Subscriber subscriber) {
-        if (others instanceof Collection) {
-            return new FixedCountMergeSubscriber(subscriber, 1 + ((Collection) others).size(), delayError);
-        } else {
-            return new DynamicCountSubscriber(subscriber, delayError);
-        }
+        return new DynamicCountSubscriber(subscriber, delayError);
```

Comment:
Why not to keep pre-existing check for `Collection`?

Reply:
there is not longer a meaningful different between the two implementations, and some collections `size()` is costly and requires iteration.

Reply:
Agreed, difference is negligible. Saying that, do we need 2 impls or can we always use dynamic one?

Reply:
Sorry for back and forth. Existing code lgtm as well.

Reply:
np. thanks for review!